src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -263,6 +263,8 @@ internal sealed override IEnumerable<Cci.INestedTypeDefinition> GetSynthesizedNe
                                         //       but adds synthesized instance constructors, Roslyn adds both
                                         var method = (MethodSymbol)member;
                                         if (method.IsDefaultValueTypeConstructor() ||
+                                            // Synthesized static constructors that will not be emitted should not be included in the SymbolToLocationMap
+                                            (method.IsImplicitStaticConstructor && GetMethodBody(method) is null) ||
```

Comment:
>(method.IsImplicitStaticConstructor && GetMethodBody(method) is null) || [](start = 44, length = 72)

I am investigating a separate issue and this loop is causing similar issue, i.e. it is duplicates what Get...ToEmit APIs do and adjustments to them causing breaks here unless the logic is duplicated. I think we should rewrite it in terms of those APIs.